WordPress/gutenberg
# Update the Gutenberg plugin to require at least the WP 6.1 version

## What?

WordPress 6.2 was released a month ago. It's time to bump the minimum required version for the Gutenberg plugin and remove the compat layer.

It also fixes #45870.

## How?
* I removed all the files from the `compat/wordpress-6.1` directory.
* Fixed all the PHP notices and fatal errors.
* Updated client-asset.php to handle preferences API ([7337428](https://github.com/WordPress/gutenberg/pull/50079/commits/73374289c743626f1948139a68b7e4d9cc28294e)). Props to @talldan for leaving detailed instructions 🦸

## Testing Instruction

* Run the Gutenberg plugin with WP 6.1, and confirm editors work as before.
* Run the Gutenberg plugin with WP 6.2, and confirm editors work as before.
